File: ccif/src/java/other/org/himalay/ccif/FirmwareVerReportS_Mode.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:05 EST 2016")

public  class FirmwareVerReportS_Mode extends BinStruct implements PublicBinMsg{




    // members variables
    // firmwareVersion
    public int firmwareVersion ;
    // fwareDate
    public FirmWareDate fwareDate ;

        public FirmwareVerReportS_Mode () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize firmwareVersion
        
        // Initialize fwareDate
        fwareDate = new FirmWareDate();
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read firmwareVersion
        {firmwareVersion=istream.readUnsignedShort(); retVal+=2;}
        // read fwareDate
        retVal += fwareDate.read(istream);

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        { /** fix dependent sizes for fwareDate **/  }
    
                // write firmwareVersion
        ostream.writeShort(firmwareVersion); retVal +=2;
        // write fwareDate
        if (fwareDate!=null)retVal +=fwareDate.write(ostream);
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("FirmwareVerReportS_Mode\n");
    dc.increaseIndent();
        int retVal= 0;
        // write firmwareVersion
        dc.indent();dc.getPs().println("firmwareVersion="+firmwareVersion+"(0x"+ Integer.toHexString(firmwareVersion) + ")" );
        // write fwareDate
        if ( fwareDate != null ) {dc.indent();dc.getPs().println("fwareDate") ;retVal +=fwareDate.dump(dc);}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for firmwareVersion
    //public int getFirmwareVersion()
    //{
    //    return firmwareVersion ;
    //}

    
    // Setter for firmwareVersion
    //public void setFirmwareVersion(int val)
    //{
    //    this.firmwareVersion= val;
    //}
    // Getter for fwareDate
    //public FirmWareDate getFwareDate()
    //{
    //    return fwareDate ;
    //}

    
    // Setter for fwareDate
    //public void setFwareDate(FirmWareDate val)
    //{
    //    this.fwareDate= val;
    //}


    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/FirmwareVerReportS_Mode.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "FirmwareVerReportS_Mode.h"

#ifdef org_himalay_ccif_FirmwareVerReportS_Mode__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
FirmwareVerReportS_Mode::FirmwareVerReportS_Mode() {
    references= 0;


    // firmwareVersion
    firmwareVersion= (ui16)0;
    // fwareDate
    org_himalay_ccif_FirmWareDate__NEW( fwareDate);
}
// Destructor
FirmwareVerReportS_Mode::~FirmwareVerReportS_Mode() {
    // firmwareVersion

    // fwareDate
    org_himalay_ccif_FirmWareDate__DELETE( fwareDate);
}

int FirmwareVerReportS_Mode::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // firmwareVersion
    {
        firmwareVersion=istream.readUI16();
        retVal+=2;
    }
    // fwareDate
    retVal += fwareDate->read(istream);
    return retVal;
}

int FirmwareVerReportS_Mode::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read firmwareVersion
    {
        firmwareVersion=istream.readUI16();
        retVal+=2;
    }
    // read fwareDate
    retVal += fwareDate->read(istream);

    return retVal;
}

int FirmwareVerReportS_Mode::write(DataOutputStream& ostream) { // throws IOException

    {   /** fix dependent sizes for fwareDate **/
    }

    int retVal= 0;
    // write firmwareVersion
    ostream.writeUI16(firmwareVersion);
    retVal +=2;
    // write fwareDate
    if ( fwareDate != NULL ) {
        retVal +=fwareDate->write(ostream);
    }

    return retVal;
}

int FirmwareVerReportS_Mode::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("FirmwareVerReportS_Mode\n");
    dc.increaseIndent();
    int retVal= 0;
    // write firmwareVersion
    dc.indent();
    dc.getPs().print("firmwareVersion=");
    dc.getPs().println((long)firmwareVersion);
    // write fwareDate
    if ( fwareDate != NULL ) {
        dc.indent();
        dc.getPs().println("fwareDate") ;
        retVal +=fwareDate->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for firmwareVersion
//ui16 FirmwareVerReportS_Mode::getFirmwareVersion()
//{
//return this->firmwareVersion;
//}

// Setter for firmwareVersion
//void FirmwareVerReportS_Mode::setFirmwareVersion(ui16 val)
//{
//this->firmwareVersion= val;
//}

// Getter for fwareDate
//org_himalay_ccif_FirmWareDate__PTR_TYPE FirmwareVerReportS_Mode::getFwareDate()
//{
//return this->fwareDate;
//}

// Setter for fwareDate
//void FirmwareVerReportS_Mode::setFwareDate(org_himalay_ccif_FirmWareDate__PTR_TYPE val)
//{
//this->fwareDate= val;
//}

int FirmwareVerReportS_Mode::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_FirmwareVerReportS_Mode__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::FirmwareVerReportS_Mode* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::FirmwareVerReportS_Mode* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code